datafusion/sql/src/unparser/expr.rs

```diff
@@ -1586,7 +1586,7 @@ impl Unparser<'_> {
                 not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
             }
             DataType::Interval(_) => {
-                not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
+                Ok(ast::DataType::Interval)
```

Comment:
I've added this as well, it seems to be a bit of a trivial conversion so not sure a test is warranted